code/frameworks/preact-vite/src/preset.ts

```diff
@@ -0,0 +1,21 @@
+import type { StorybookConfig } from '@storybook/builder-vite';
+import { hasVitePlugins } from '@storybook/builder-vite';
+import preact from '@preact/preset-vite';
+
+export const core: StorybookConfig['core'] = {
+  builder: '@storybook/builder-vite',
+  renderer: '@storybook/preact',
+};
+
+export const viteFinal: StorybookConfig['viteFinal'] = async (config) => {
+  const { plugins = [] } = config;
+
+  // Add Preact plugin if not present
+  if (!(await hasVitePlugins(plugins, ['vite:preact-jsx']))) {
+    plugins.push(preact());
+  }
+
+  // TODO: Add docgen plugin per issue https://github.com/storybookjs/storybook/issues/19739
```

Comment:
Oh I see this now, ignore my previous question about docgen, but we should still remove those dependencies.